cleverhans/model_abs.py

```diff
@@ -0,0 +1,73 @@
+from abc import ABCMeta
+from keras.models import Model
+
+
+class ModelAbstraction:
+    """
+    An abstract interface for a model that exposes the required functionalities
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, model):
+        """
+        :param model: A function that takes a symbolic input and returns the
+                      symbolic output for the model's predictions.
+        """
+        raise NotImplementedError('Model abstraction not implementated.')
+
+    def get_hidden(self, layer):
```

Comment:
Are you suggesting that the methods do not return a symbolic representation but rather compute and return the output given an input?